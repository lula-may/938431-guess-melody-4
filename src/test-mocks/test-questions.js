const questions = [
  {
    type: `singer`,
    song: {
      singer: `Lady Gaga`,
      src: `https://music/Lady_Gaga.ogg`
    },
    answers: [
      {
        singer: `Lady Gaga`,
        avatar: `https://images/Lady_Gaga.jpg`,
        id: `0`
      },
      {
        singer: `Witney Houston`,
        avatar: `https://images/Witnew_Houston.jpg`,
        id: `1`
      },
      {
        singer: `Cher`,
        avatar: `https://images/Cher.jpg`,
        id: `2`
      }
    ]
  },
  {
    type: `genre`,
    genre: `jazz`,
    answers: [
      {
        src: `https://samples/1.ogg`,
        genre: `blues`,
        id: `3`
      },
      {
        src: `https://samples/2.ogg`,
        genre: `jazz`,
        id: `4`
      },
      {
        src: `https://samples/3.ogg`,
        genre: `folk`,
        id: `5`
      },
      {
        src: `https://samples/4.ogg`,
        genre: `jazz`,
        id: `6`
      }
    ]
  }
];

export {questions};
