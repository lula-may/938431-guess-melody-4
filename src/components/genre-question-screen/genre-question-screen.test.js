import React from "react";
import renderer from "react-test-renderer";
import GenreQuestionScreen from "./genre-question-screen.jsx";
import {questions} from "../../test-mocks/test-questions";

const question = questions[1];

it(`Should correctly render GenreQuestionScreen`, () => {
  const tree = renderer.create(
      <GenreQuestionScreen
        onAnswer={() => {}}
        question={question}
        renderPlayer={() => {}}
      />, {
        createNodeMock: () => {
          return {};
        }
      }
  )
  .toJSON();

  expect(tree).toMatchSnapshot();
});
