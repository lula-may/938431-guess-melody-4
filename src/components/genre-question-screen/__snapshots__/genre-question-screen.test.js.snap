// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render GenreQuestionScreen 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Выберите 
    jazz
     треки
  </h2>
  <form
    className="game__tracks"
    onSubmit={[Function]}
  >
    <div
      className="track"
    >
      <div
        className="game__answer"
      >
        <input
          checked={false}
          className="game__input visually-hidden"
          id="answer-3"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="answer-3"
        />
        <label
          className="game__check"
          htmlFor="answer-3"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <div
        className="game__answer"
      >
        <input
          checked={false}
          className="game__input visually-hidden"
          id="answer-4"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="answer-4"
        />
        <label
          className="game__check"
          htmlFor="answer-4"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <div
        className="game__answer"
      >
        <input
          checked={false}
          className="game__input visually-hidden"
          id="answer-5"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="answer-5"
        />
        <label
          className="game__check"
          htmlFor="answer-5"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <div
        className="game__answer"
      >
        <input
          checked={false}
          className="game__input visually-hidden"
          id="answer-6"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="answer-6"
        />
        <label
          className="game__check"
          htmlFor="answer-6"
        >
          Отметить
        </label>
      </div>
    </div>
    <button
      className="game__submit button"
      type="submit"
    >
      Ответить
    </button>
  </form>
</section>
`;
